### PR TITLE
Add prop to force fullscreen preview for videos.

### PR DESCRIPTION
**Pull request by SergioEstevao**
## Description

Added a prop to force fullscreen preview for videos.
On Android, by default, we force fullscreen mode to allow usersto watch videos using outside players.
On iOS the fullscreen is not forced by default to allow videos to displayed inline or fullscreen depending of the user actions.

## How has this been tested?

This can be tested using the GB-mobile PR here: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1952
## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/video-player/index.native.js [hunk 1]
```diff
@@ -84,7 +85,10 @@ class Video extends Component {
 	}
 
 	render() {
-		const { isSelected, style } = this.props;
+		const { isSelected, style, forceFullscreen } = this.props;
+		if ( forceFullscreen !== undefined ) {
+			this.forceFullscreen = forceFullscreen;
+		}
```
Comment (etoledom) [inline packages/block-editor/src/components/video-player/index.native.js]: Maybe using a default value is cleaner? In this way we don't need to declare `this.forceFullscreen`
```suggestion
		const { isSelected, style, forceFullscreen = ! this.isIOS } = this.props;
```

**Review by etoledom — COMMENTED** [review]
Thanks @SergioEstevao for this PR 🎉

I noticed one difference on iOS:
When I tap on the unselected video block to select it, the video will automatically start playing. As far as I remember, we needed to select the block first and then tap again to play the video if we wanted to.

Is this change intentional? 

video_block_ios

One other interesting thing that happened on Android is that when the external browser opens, it will ask me to download the video instead of playing it.

This happened with Chrome and DuckDuckGo. Testing with Firefox it properly played it without the need to download it. 

I was able to reproduce this same problem on develop so probably is not part of this PR.
Still worth checking it out

cc @pinarol @iamthomasbishop

**Comment by iamthomasbishop** [comment]
> When I tap on the unselected video block to select it, the video will automatically start playing. As far as I remember, we needed to select the block first and then tap again to play the video if we wanted to.

@etoledom That's correct. The first tap should select the block, then a second tap to play the media. Same thing w/ an image block — first tap selects the block, second tap views the image.

**Comment by SergioEstevao** [comment]
@etoledom I fixed the issue you detected and simplified the code, do you mind giving it another look?

**Review by etoledom — COMMENTED** [review]
Thanks for the updates @SergioEstevao !

I still see the behaviour described previously.
When the video block is not selected and I tap on the Play button, the block gets selected and the video starts playing automatically. It doesn't happen if I tap outside of the Play button.

Adding this prop to the `VideoPlayer` component fixed it for me:
`pointerEvents={ isSelected ? undefined : "none" }`

**Comment by SergioEstevao** [comment]
@etoledom I followed your suggestion for tapping on the play button. Can you please give this another check?

**Review by etoledom — APPROVED** [review]
Working great now! 🎉 
Thank you for the update

**Pull request merged**